kld/src/cli/client.rs

```diff
@@ -273,6 +273,16 @@ impl Api {
         deserialize::<Vec<Payment>>(response)
     }
 
+    pub fn lsp_list_protocols(&self, node_id: Option<PublicKey>) -> Result<String> {
+        let route = if let Some(node_id) = node_id {
+            format!("{}?node_id={node_id}", routes::LSP_LIST_PROTOCOLS)
```

Comment:
you don't need to do this manually. RequestBuilder has a query function you can pass the params to.